### PR TITLE
OP-118 module from o to p

### PR DESCRIPTION
**Pull request by gildasdongmo**

### CODE REVIEW

**Comment by mwithi** [comment]
Guys do you think we can merge this and close OP-6 finally?

Cc: @vicusotto @emecas

**Review by emecas — CHANGES_REQUESTED** [review]
I would start removing those huge logs files.

File: src/main/java/org/isf/patvac/rest/PatVacController.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	 */
+	@PostMapping(value = "/patientvaccines", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<Integer> newPatientVaccine(@RequestBody PatientVaccineDTO patientVaccineDTO) throws OHServiceException {
+		int code = patientVaccineDTO.getCode();
+		logger.info("Create patient vaccine " + code);
+		boolean isCreated = patVacManager.newPatientVaccine(mapper.map2Model(patientVaccineDTO));
+		if (!isCreated) {
+			throw new OHAPIException(new OHExceptionMessage(null, "patient vaccine is not created!", OHSeverityLevel.ERROR));
+		}
+		return ResponseEntity.status(HttpStatus.CREATED).body(code);
```
Comment (vicusotto) [inline src/main/java/org/isf/patvac/rest/PatVacController.java:67]: Is the DTO code the same that will be insert?

File: src/main/java/org/isf/patvac/rest/PatVacController.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	/**
+	 * Updates the specified {@link PatientVaccine}.
+	 * @param patientVaccineDTO
+	 * @return <code>true</code> if the operation type has been updated, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/patientvaccines", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<Integer> updatePatientVaccinet(@RequestBody PatientVaccineDTO patientVaccineDTO)
```
Comment (vicusotto) [inline src/main/java/org/isf/patvac/rest/PatVacController.java:77]: Put Method should have code as @PathVariable. I know it should be an http standard.
`@PutMapping(value = "/patientvaccines/{code}", produces = MediaType.APPLICATION_JSON_VALUE)`

File: src/main/java/org/isf/patvac/rest/PatVacController.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	 * @param code
+	 * @return <code>true</code> if the {@link PatientVaccine} has been deleted, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@DeleteMapping(value = "/patientvaccines/{code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> deletePatientVaccine(@PathVariable int code) throws OHServiceException {
+		logger.info("Delete patient vaccine code:" + code);
+		PatientVaccine patVac = new PatientVaccine();
+		patVac.setCode(code);
+		boolean isDeleted = patVacManager.deletePatientVaccine(patVac);
```
Comment (vicusotto) [inline src/main/java/org/isf/patvac/rest/PatVacController.java:152]: Does this delete work regardless the lock column @Version?

File: src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	@PostMapping(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> newPregnantTreatmentType(@RequestBody PregnantTreatmentTypeDTO pregnantTreatmentTypeDTO) throws OHServiceException {
+		String code = pregnantTreatmentTypeDTO.getCode();
+		logger.info("Create pregnant treatment Type " + code);
+		boolean isCreated = pregTreatTypeManager.newPregnantTreatmentType(mapper.map2Model(pregnantTreatmentTypeDTO));
+		PregnantTreatmentType pregTreatTypeCreated = null;
+		List<PregnantTreatmentType> pregTreatTypeFounds = pregTreatTypeManager.getPregnantTreatmentType().stream().filter(pregtreattype -> pregtreattype.getCode().equals(code))
+				.collect(Collectors.toList());
+		if (pregTreatTypeFounds.size() > 0)
+			pregTreatTypeCreated = pregTreatTypeFounds.get(0);
```
Comment (vicusotto) [inline src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java:62]: List could be avoided:
`		PregnantTreatmentType pregTreatTypeCreated = pregTreatTypeManager.getPregnantTreatmentType().stream().filter(pregtreattype -> pregtreattype.getCode().equals(code))
				.findFirst().orElse(null);`

File: src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	/**
+	 * Updates the specified {@link PregnantTreatmentType}.
+	 * @param pregnantTreatmentTypeDTO
+	 * @return <code>true</code> if the pregnant treatment type has been updated, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> updatePregnantTreatmentTypet(@RequestBody PregnantTreatmentTypeDTO pregnantTreatmentTypeDTO)
```
Comment (vicusotto) [inline src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java:76]: Put Method should have code as @PathVariable. I know it should be an http standard.
`@PutMapping(value = "/pregnanttreatmenttypes/{code}", produces = MediaType.APPLICATION_JSON_VALUE)`

File: src/main/java/org/isf/priceslist/rest/PriceListController.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+	}
+
+	/**
+	 * Updates the specified {@link PriceList}.
+	 * @param priceListDTO
+	 * @return <code>true</code> if the price list has been updated, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/pricelists", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> updatePriceListt(@RequestBody PriceListDTO priceListDTO)
```
Comment (vicusotto) [inline src/main/java/org/isf/priceslist/rest/PriceListController.java:77]: Put Method should have code as @PathVariable. I know it should be an http standard.
`@PutMapping(value = "/pricelists/{id}", produces = MediaType.APPLICATION_JSON_VALUE)`

File: src/main/java/org/isf/pricesothers/rest/PricesOthersController.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	/**
+	 * Updates the specified {@link PricesOthers}.
+	 * @param pricesOthersDTO
+	 * @return <code>true</code> if the prices others has been updated, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/pricesothers", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> updatePricesOtherst(@RequestBody PricesOthersDTO pricesOthersDTO)
```
Comment (vicusotto) [inline src/main/java/org/isf/pricesothers/rest/PricesOthersController.java:70]: Put Method should have code as @PathVariable. I know it should be an http standard.
`@PutMapping(value = "/pricesothers/{id}", produces = MediaType.APPLICATION_JSON_VALUE)`

**Review by vicusotto — COMMENTED** [review]
Thanks for your work @gildasdongmo 
I left a few comments to your PR.

**Comment by vicusotto** [comment]
> Guys do you think we can merge this and close [OP-6](https://openhospital.atlassian.net/browse/OP-6) finally?
> 
> Cc: @vicusotto @emecas

Hi @mwithi 
I did my review and left some comments.

**Comment by mwithi** [comment]
> I would start removing those huge logs files.

Done!

**Review by mwithi — CHANGES_REQUESTED** [review]
Hi @gildasdongmo, could you please verify the suggestions from @vicusotto ?

This is the last PRs for OP-6 and we are all thrillred to let it land on develop soon 😄

Also, I don't know why checks are failing, on my local `maven clean install` works

Thanks for your work!

**Comment by gildasdongmo** [comment]
Hi All, and sorry for the silence,
thank you @vicusotto for the review, I will apply the corrections.
@mwithi my too I don't know why the checks are failing, I'll take a close look at it.

File: pom.xml [hunk 8]
```diff
@@ -115,6 +115,13 @@
 			<artifactId>javax.el</artifactId>
 			<version>2.2.4</version>
 		</dependency>
+		<!-- https://mvnrepository.com/artifact/commons-logging/commons-logging -->
+		<dependency>
+		    <groupId>commons-logging</groupId>
+		    <artifactId>commons-logging</artifactId>
+		    <version>1.2</version>
+		</dependency>
+
```
[the remark this answers is not in the document]
Comment (gildasdongmo) [reply pom.xml:124]: ok my repository was not up to date.

File: src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java (hunk 4 above)
Comment (gildasdongmo) [reply src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java:62]: ok

File: src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java (hunk 5 above)
Comment (gildasdongmo) [reply src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java:76]: the code is already in the dto, should I add it again as @PathVariable ?

**Comment by emecas** [comment]
> @mwithi my too I don't know why the checks are failing, I'll take a close look at it.

@gildasdongmo 

Checks are failing due to a compilation error on `VaccineTypeController`

image

Because your branch `uni2grow:op-118-module-from-O-to-P` is missing the changes from PR #65 (e3c0ddf57e2)

image

**Comment by mwithi** [comment]
Sorry @gildasdongmo, I've added the changes suggested by @vicusotto, test are fine, and also checks are fine now.

I'm going to merge this. Thanks everybody!!!

**Review by mwithi — APPROVED** [review]

**Pull request merged**